File: covilo-core/src/main/java/me/kvdpxne/covilo/infrastructure/security/encoding/PasswordEncoderConfiguration.java
package me.kvdpxne.covilo.infrastructure.security.encoding;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder.BCryptVersion;
import org.springframework.security.crypto.password.PasswordEncoder;

@Configuration
public class PasswordEncoderConfiguration {

  @Bean
  public PasswordEncoder passwordEncoder() {
    return new BCryptPasswordEncoder(BCryptVersion.$2Y);
  }
}

File: covilo-core/src/main/kotlin/me/kvdpxne/covilo/infrastructure/security/PasswordEncoderConfiguration.kt
package me.kvdpxne.covilo.infrastructure.security

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

@Configuration
class PasswordEncoderConfiguration {

  @Bean(name = ["passwordEncoder"])
  fun getPasswordEncoder(): PasswordEncoder = BCryptPasswordEncoder()
}